Core/clim-core/menu.lisp

```diff
@@ -192,8 +196,17 @@ account, and create a list of menu buttons."
 	  (create-substructure sub-menu sub-menu)))
     (arm-menu sub-menu)))
 
-(defmethod handle-event ((pane menu-button-submenu-pane) (event pointer-button-release-event))
-  (destroy-substructure (menu-root pane)))
+(defmethod handle-event ((pane menu-button-submenu-pane) (event pointer-button-press-event))
+  (if (slot-value pane 'armed)
+      (destroy-substructure pane)
+      (arm-branch pane)))
+
+(defmethod handle-event :before (pane (event pointer-button-release-event))
```

Comment:
why isn't this specialized on pane as `menu-button-pane`?

Reply:
then handle-event would trigger only for panes of this type and there would be no unnecessary overhead

Reply:
ah, nevermind.